File: BusTicketBookingSystem/src/main/java/com/ash/bbus/web/BusTicketBookingSystem/service/TripService.java
package com.ash.bbus.web.BusTicketBookingSystem.service;

import com.ash.bbus.web.BusTicketBookingSystem.dto.TripDTO;
import java.time.LocalDate;
import java.util.List;

public interface TripService {

    TripDTO createTrip(TripDTO dto);
    TripDTO getTripById(Long id);
    List<TripDTO> getAllTrips();

    // ✅ Search trips — used by frontend search page
    List<TripDTO> searchTrips(String source, String destination,
                              LocalDate travelDate);

    TripDTO updateTrip(Long id, TripDTO dto);
    void cancelTrip(Long id);
}

File: bus-booking-frontend/src/services/tripService.js
import api from "../api/axiosInstance";

export const searchTrips = (source, destination, date) =>
  api.get("/trips/search", {
    params: { source, destination, date },
  });

export const getTripById = (id) =>
  api.get(`/trips/${id}`);

export const getAllTrips = () =>
  api.get("/trips");

export const createTrip = (data) =>
  api.post("/trips", data);

export const updateTrip = (id, data) =>
  api.put(`/trips/${id}`, data);

export const cancelTrip = (id) =>
  api.put(`/trips/${id}/cancel`);